R/driver-sql-server.R

```diff
@@ -81,6 +81,32 @@ setMethod("dbExistsTable", c("Microsoft SQL Server", "character"),
   }
 )
 
+#' @description
+#' ## `dbListTables()`
+#' The default implementation reports temporary tables as non-existent
+#' when a `catalog_name` isn't supplied since they live in a different catalog.
+#' This method provides a special case for temporary tables.
+#' @rdname SQLServer
+#' @usage NULL
+setMethod("dbListTables", "Microsoft SQL Server",
+  function(conn,
+           catalog_name = NULL,
+           schema_name = NULL,
+           table_name = NULL,
+           table_type = NULL,
+           ...) {
+    res <- callNextMethod()
+
+    if (is.null(catalog_name)) {
```

Comment:
```suggestion
    if (is.null(catalog_name) && is.null(schema_name)) {
```

Since IIUC temp tables live in their own schema too

Reply:
Some thoughts here in https://github.com/r-dbi/odbc/pull/747#discussion_r1459638726—I'm on board if we indeed want to exclude those other schemas though.

Reply:
I think we should do both maybe? I'm not 100% sure what people would expect here, but given that temporary tables exist in their own catalog + schema, I think if you request a specific catalog/schema, we should respect that.